### DISCUSSION TITLE
Popup scrollbar not working on iOS

### DISCUSSION THREAD

**Discussion by gem0303** [KingSora/OverlayScrollbars#546] [Q&A] [answered]
Reduced test case Codepen running OS v2.2.1: https://codepen.io/gem0303/pen/dyQQNxg

I have a project with two Overlay scrollbars -- one for the main content (var scrollInstance), and a separate one for a popup (var popup_scrollbar). The popup scrollbar works fine on desktop on all major browsers, but it seems to have a bug on iOS on Safari and Chrome. The scrollbar does not appear the first time I open the popup. However, it does appear if I close the popup and reopen it. It also works if I set "showNativeOverlaidScrollbars: true" but at that point the plugin is just completely disabled, correct?

edit to add, OS v1.13.0 works fine on all desktop and mobile browsers. Codepen link: https://codepen.io/gem0303/pen/YzRRVqq

Please let me know if you have any insight into what could be causing this issue.

**Comment by KingSora** [comment, marked as answer]
Good day @gem0303 

That was a longer debugging session for me.. aaaand I figured out why that happens! Usually the `ResizeObserver` is always triggered when its created and connected. In Webkit this is not the case if the element is hidden in that point in time. I had a small optimization in place which would always skip the first `ResizeObserver` callback. Thats also the reason you had to open the pupup twice.

I've fixed the bug and will publish it with the next release :)

Here is the fixed version: https://codepen.io/kingsora-the-decoder/pen/jOQQdQe If you can please try it out and report back whether it works for you

**Reply by gem0303** [reply]
Works perfectly! Thank you for investigating and solving so quickly.
